apis/refs/v1beta1/projectref.go

```diff
@@ -60,6 +60,9 @@ type Project struct {
 // ResolveProject will resolve a ProjectRef to a Project, with the ProjectID.
 func ResolveProject(ctx context.Context, reader client.Reader, src client.Object, ref *ProjectRef) (*Project, error) {
 	if ref == nil {
+		if projectID := src.GetAnnotations()["cnrm.cloud.google.com/project-id"]; projectID != "" {
+			return &Project{ProjectID: projectID}, nil
+		}
```

Comment:
Personally I'd prefer having this `project-id` annotation being handled in each individual KCC resource because this is more of a backward compatible feature and is not what we recommend developers to use (we recommend `spec.projectRef`). WDYT?

Reply:
So the challenge is that it's pretty easy to forget, and we might not have test coverage to catch it!

Have we (aka you :-) ) written the "don't use project-id" guidance?

Personally I think we should continue to support the "fallback to namespace name" functionality (which is what is currently implemented by the project-id annotation, though we could do it another way).  Where we have included projectRef, we have made it mandatory.  I actually disagree with that, because it breaks users' ability to have the k8s namespace == the project-id and not have to specify projects everywhere.

But I will move this so that we can reach a full decision / direction here, rather than sneaking it in an unrelated PR :-)

cc @cheftako

Reply:
"k8s namespace == the project-id" is nit. My concerns is that KCC have already built several conventions but does not give a complete story and users cannot purely based on the KCC conventions to guess out the correct answers. i.e. If using project-id, users cannot use it as a no-brainer, they need to check if each resource supports that annotation or not. For `spec.projectRef`, at least users can rely on the  CRD validation to make some reasonable guess and self-correct.